cmd/ethermintd/genaccounts.go

```diff
@@ -48,58 +44,66 @@ contain valid denominations. Accounts may optionally be supplied with vesting pa
 `,
 		Args: cobra.ExactArgs(2),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			config := ctx.Config
-			config.SetRoot(viper.GetString(cli.HomeFlag))
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			depCdc := clientCtx.JSONMarshaler
+			cdc := depCdc.(codec.Marshaler)
+
+			serverCtx := server.GetServerContextFromCmd(cmd)
+			config := serverCtx.Config
+
+			config.SetRoot(clientCtx.HomeDir)
 
 			addr, err := sdk.AccAddressFromBech32(args[0])
-			inBuf := bufio.NewReader(cmd.InOrStdin())
 			if err != nil {
-				// attempt to lookup address from Keybase if no address was provided
-				kb, err := keys.NewKeyring(
+				inBuf := bufio.NewReader(cmd.InOrStdin())
+				keyringBackend, _ := cmd.Flags().GetString(flags.FlagKeyringBackend)
+
+				// attempt to lookup address from Keyring if no address was provided
+				kr, err := keyring.New(
 					sdk.KeyringServiceName(),
-					viper.GetString(flags.FlagKeyringBackend),
-					viper.GetString(flagClientHome),
+					keyringBackend,
+					clientCtx.HomeDir,
 					inBuf,
-					hd.EthSecp256k1Options()...,
+					hd.EthSecp256k1Option(),
 				)
 				if err != nil {
 					return err
 				}
 
-				info, err := kb.Get(args[0])
+				// TODO: fix error, keyring.Info
```

Comment:
what is the error here?

Reply:
I think this is a leftover comment, will remove